pkg/apis/core/types.go

```diff
@@ -4078,10 +4078,19 @@ type ServicePort struct {
 	Protocol Protocol
 
 	// The application protocol for this port.
+	// This is used as a hint for implementations to offer richer behavior for protocols that they understand.
 	// This field follows standard Kubernetes label syntax.
-	// Un-prefixed names are reserved for IANA standard service names (as per
+	// Valid values are either:
+	//
+	// * Un-prefixed protocol names - reserved for IANA standard service names (as per
 	// RFC-6335 and https://www.iana.org/assignments/service-names).
-	// Non-standard protocols should use prefixed names such as
+	//
+	// * Kubernetes-defined prefixed names:
+	//   * 'kubernetes.io/h2c' - http2 over cleartext as described in https://www.rfc-editor.org/rfc/rfc7540
+	//   * 'kubernetes.io/grpc' - grpc traffic - see https://github.com/grpc/grpc/blob/v1.51.1/doc/PROTOCOL-HTTP2.md
+	//   * 'kubernetes.io/tcp' - plain tcp traffic
```

Comment:
Copying from lost comment:

so we're asking people to specify "TCP" twice in order to defend against implementations? That seems...unpleasant.

If we need to add a hint here, can we call it "raw" or something, which works for all port-protocols and tells implementations "don't try to be clever".

Reply:
Had an offline chat with @robscott . I removed it from here as it probably worth a little bit more discussion after we convert it to raw for all port protocols.

I will amend the KEP accordingly and add it as a Follow-up.

I know from [this](https://github.com/kubernetes/enhancements/pull/3727#discussion_r1068613722) comment that it is probably useful to have such "raw" value but I think we better iterate here than squeeze undesired change on last minute.